server/services/auth_service.py

```
@@ -56,25 +59,25 @@ async def login(self, login_data: LoginRequest) -> LoginResponse:
+        try:
+            self.auth_domain.validate_login_data(login_data)
+            user_id = await self.auth_repository.verify_password(login_data.email, login_data.password)
+            user = await self.auth_repository.get_user_by_id(user_id)
+            self.auth_domain.validate_user_exists(user)
+            token = await self.auth_repository.create_custom_token(user.user_id)
+        except InputValidationError:
+            raise
+        except AuthError:
+            raise
+        except KeyError:
+            raise DependencyUnavailableError("Firebase API key not configured")
```

Ruff (0.14.3)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)